pyccl/emulators/baccoemu_linear_pk.py

```diff
@@ -0,0 +1,118 @@
+        import warnings
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=UserWarning)
+            import baccoemu
+            self.mpk = baccoemu.Matter_powerspectrum()
+        self.a_min = self.mpk.emulator['linear']['bounds'][-1][0]
+        self.a_max = self.mpk.emulator['linear']['bounds'][-1][1]
+        self.k_min = self.mpk.emulator['linear']['k'][0]
+        self.k_max = self.mpk.emulator['linear']['k'][-1]
+
+    def __str__(self) -> str:
+        return """baccoemu linear Pk module,
+k_min,k_max = ({}, {}),
+a_min,a_max = ({}, {})""".format(
+            self.k_min, self.k_max, self.a_min, self.a_max)
+
+    def _sigma8tot_2_sigma8cold(self, emupars, sigma8tot):
+        """Use baccoemu to convert sigma8 total matter to sigma8 cdm+baryons
+        """
+        if hasattr(emupars['omega_cold'], '__len__'):
+            _emupars = {}
+            for pname in emupars:
+                _emupars[pname] = emupars[pname][0]
+        else:
+            _emupars = emupars
+        A_s_fid = 2.1e-9
+        sigma8tot_fid = self.mpk.get_sigma8(cold=False, A_s=A_s_fid,
+                                            **_emupars)
+        A_s = (sigma8tot / sigma8tot_fid)**2 * A_s_fid
+        return self.mpk.get_sigma8(cold=True, A_s=A_s, **_emupars)
+
+    def _get_pk_at_a(self, a, cosmo):
+        # First create the dictionary passed to baccoemu
+        # if a is an array, make sure all the other parameters passed to the
+        # emulator have the same len
+        if hasattr(a, '__len__'):
+            emupars = dict(
+                omega_cold=np.full((len(a)),
+                                   cosmo['Omega_c'] + cosmo['Omega_b']),
+                omega_baryon=np.full((len(a)), cosmo['Omega_b']),
+                ns=np.full((len(a)), cosmo['n_s']),
+                hubble=np.full((len(a)), cosmo['h']),
+                neutrino_mass=np.full((len(a)), np.sum(cosmo['m_nu'])),
+                w0=np.full((len(a)), cosmo['w0']),
+                wa=np.full((len(a)), cosmo['wa']),
+                expfactor=a
+            )
+        else:
+            emupars = dict(
+                omega_cold=cosmo['Omega_c'] + cosmo['Omega_b'],
+                omega_baryon=cosmo['Omega_b'],
+                ns=cosmo['n_s'],
+                hubble=cosmo['h'],
+                neutrino_mass=np.sum(cosmo['m_nu']),
+                w0=cosmo['w0'],
+                wa=cosmo['wa'],
+                expfactor=a
+            )
+
+        # if cosmo contains sigma8, we use it for baccoemu, otherwise we pass
+        # A_s to the emulator
+        if np.isnan(cosmo['A_s']):
+            sigma8tot = cosmo['sigma8']
+            sigma8cold = self._sigma8tot_2_sigma8cold(emupars, sigma8tot)
+            if hasattr(a, '__len__'):
+                emupars['sigma8_cold'] = np.full((len(a)), sigma8cold)
+            else:
+                emupars['sigma8_cold'] = sigma8cold
+        else:
```

Comment:
I think the benchmark tests don't check the case when you actually have A_s in the cosmo parameters. Can you add a quick test for this? It could be something where you create two cosmologies, one with A_s and one with the corresponding sigma8, and you check that this returns compatible P(k)s (within numerical errors), for instance.

Reply:
Do this for the nonlinear bacco too.

Reply:
Done